lib/ansible/plugins/connection/kubernetes.py

```diff
@@ -0,0 +1,328 @@
+               - section: defaults
+                 key: remote_user
+            env_vars:
+               - ANSIBLE_REMOTE_USER
+            host_vars:
+               - ansible_user
+'''
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+import os
+import time
+import sys
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils.six.moves import shlex_quote
+from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.plugins.connection import ConnectionBase, BUFSIZE
+
+# NOTE: This was copied from inventory/docker.py
+# Manipulation of the path is needed because the kubernetes
+# module is imported by the name kubernetes, and because this file
+# is also named kubernetes
+for path in [os.getcwd(), '', os.path.dirname(os.path.abspath(__file__))]:
+    try:
+        del sys.path[sys.path.index(path)]
+    except:
+        pass
+try:
+    from kubernetes import config
+    from kubernetes.config import ConfigException
+    from kubernetes.config.kube_config import KubeConfigLoader
+    from kubernetes.client import configuration
+    from kubernetes.client.apis import core_v1_api
+    from kubernetes.client.rest import ApiException
+except ImportError as ie:
+    raise AnsibleError("Kubernetes library is not installed: {0}".format(ie.message))
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+
+    display = Display()
+
+
+class Connection(ConnectionBase):
+    ''' Local based connections '''
+    transport = 'kubernetes'
+    has_pipelining = False
+    active_context = None
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
+
+        self.remote_user = None
+        self.actual_user = None
+        self.name = self._play_context.remote_addr
+        self.response = None
+        configuration.assert_hostname = False
+
+    def requested_context(self):
+        server = "{0}:{1}".format(self._play_context.kubernetes_cluster, self._play_context.kubernetes_port)
+        return "{0}/{1}/{2}".format(
+            self._play_context.kubernetes_namespace,
+            server,
+            self._play_context.remote_user,
+        )
+
+    def generate_kubernetes_config(self):
+
+        server = "{0}:{1}".format(self._play_context.kubernetes_cluster, self._play_context.kubernetes_port)
+        context = "{0}/{1}/{2}".format(
+            self._play_context.kubernetes_namespace,
+            server,
+            self._play_context.remote_user,
+        )
+        users_name = "{0}/{1}".format(
+            self._play_context.remote_user,
+            server
+        )
+
+        kubernetes_config = {
+            "current-context": context,
+            "clusters": [
+                {
+                    "cluster": {
+                        "insecure-skip-tls-verify": "true",
```

Comment:
@chrrrles your right it probably should be an option for `insecure-skip-tls-verify`.